packages/jobs/utils/contract-utils.js

```diff
@@ -1,57 +1,79 @@
-const Web3 = require('web3');
-const Promise = require('bluebird');
 const contract = require('truffle-contract');
+const ethers = require('ethers');
 
 const config = require('../config');
 const logger = require('./logger')('utils/contract-utils');
 
-let ContractUtils = {};
-
 const contractsFolder = '../build/contracts/';
-let web3 = null;
+const MAX_ITERATIONS_FOR_WALLET_SEARCHING = 10000;
+const LOG_INTERATIONS_INTERVAL = 1000;
+
+let ContractUtils = {
+  getProvider,
+};
 
-const AliceToken = initializeContract('AliceToken');
-const ClaimsRegistry = initializeContract('ClaimsRegistry');
+ContractUtils.mainWallet = getMainWallet();
+validateNetworkId(ContractUtils.mainWallet.provider);
 
-ContractUtils.getClaimsRegistry = (address) => {
-  return getContractInstance(ClaimsRegistry, address);
+ContractUtils.getTruffleContract = function (name) {
+  const artefacts = require(contractsFolder + name + '.json');
+  return contract(artefacts);
 };
 
-ContractUtils.getProvider = function () {
-  lazyInitWeb3();
-  return web3.currentProvider;
+ContractUtils.getContractInstance = function (
+  contractName,
+  address,
+  wallet=ContractUtils.mainWallet
+) {
+  const contract = ContractUtils.getTruffleContract(contractName);
+  return new ethers.Contract(address, contract.abi, wallet);
 };
 
-ContractUtils.getWeb3 = function () {
-  lazyInitWeb3();
-  validateNetworkId();
-  return web3;
+ContractUtils.getWalletForIndex = function (index) {
+  let path = `m/44'/60'/0'/0/${index}`;
+  return new ethers.Wallet.fromMnemonic(config.mnemonic, path);
 };
 
-Object.assign(ContractUtils, {
-  AliceToken,
+ContractUtils.getWallet = function (address) {
+  for (let counter = 0; counter < MAX_ITERATIONS_FOR_WALLET_SEARCHING; counter++) {
+    if (counter > 0 && counter % LOG_INTERATIONS_INTERVAL == 0) {
```

Comment:
Hmm, why do you iterate instead of using the address->index relationship stored in db?